Index: micropython/ports/esp32/boards/LILYGO_EPD47/mpconfigboard.cmake
===================================================================
--- micropython.orig/ports/esp32/boards/LILYGO_EPD47/mpconfigboard.cmake
+++ micropython/ports/esp32/boards/LILYGO_EPD47/mpconfigboard.cmake
@@ -9,6 +9,7 @@ if(NOT MICROPY_FROZEN_MANIFEST)
     set(MICROPY_FROZEN_MANIFEST
         ${MICROPY_PORT_DIR}/boards/manifest.py
         $(EXTMOD_FROZEN_DIR)/touch/manifest.py
+        $(EXTMOD_FROZEN_DIR)/display/framebuf1/manifest.py
     )
 endif()
 
