Index: LilyGo-EPD47/src/ed097oc4.c
===================================================================
--- LilyGo-EPD47.orig/src/ed097oc4.c
+++ LilyGo-EPD47/src/ed097oc4.c
@@ -106,6 +106,15 @@ void epd_base_init(uint32_t epd_row_widt
   rmt_pulse_init(CKV);
 }
 
+
+void epd_base_deinit()
+{
+    while (rmt_busy()) ;
+    rmt_pulse_deinit();
+    i2s_deinit();
+}
+
+
 void epd_poweron() {
   // POWERON
   config_reg.ep_scan_direction = true;
Index: LilyGo-EPD47/src/ed097oc4.h
===================================================================
--- LilyGo-EPD47.orig/src/ed097oc4.h
+++ LilyGo-EPD47/src/ed097oc4.h
@@ -26,6 +26,7 @@
 #define D0 GPIO_NUM_33
 
 void epd_base_init(uint32_t epd_row_width);
+void epd_base_deinit();
 void epd_poweron();
 void epd_poweroff();
 
Index: LilyGo-EPD47/src/epd_driver.c
===================================================================
--- LilyGo-EPD47.orig/src/epd_driver.c
+++ LilyGo-EPD47/src/epd_driver.c
@@ -70,6 +70,15 @@ void epd_init()
     output_queue = xQueueCreate(32, EPD_WIDTH / 2);
 }
 
+
+void epd_deinit()
+{
+    epd_base_deinit();
+    free(conversion_lut);
+    vQueueDelete(output_queue);
+}
+
+
 // skip a display row
 void skip_row(uint8_t pipeline_finish_time)
 {
Index: LilyGo-EPD47/src/epd_driver.h
===================================================================
--- LilyGo-EPD47.orig/src/epd_driver.h
+++ LilyGo-EPD47/src/epd_driver.h
@@ -65,6 +65,8 @@ typedef struct {
 /** Initialize the ePaper display */
 void epd_init();
 
+void epd_deinit();
+
 /** Enable display power supply. */
 void epd_poweron();
 
Index: LilyGo-EPD47/src/rmt_pulse.c
===================================================================
--- LilyGo-EPD47.orig/src/rmt_pulse.c
+++ LilyGo-EPD47/src/rmt_pulse.c
@@ -61,6 +61,13 @@ void rmt_pulse_init(gpio_num_t pin)
 #endif
 }
 
+
+void rmt_pulse_deinit()
+{
+    rmt_driver_uninstall(row_rmt_config.channel);
+}
+
+
 void IRAM_ATTR pulse_ckv_ticks(uint16_t high_time_ticks,
                                uint16_t low_time_ticks, bool wait)
 {
Index: LilyGo-EPD47/src/rmt_pulse.h
===================================================================
--- LilyGo-EPD47.orig/src/rmt_pulse.h
+++ LilyGo-EPD47/src/rmt_pulse.h
@@ -16,6 +16,8 @@
  */
 void rmt_pulse_init(gpio_num_t pin);
 
+void rmt_pulse_deinit();
+
 /**
  * Outputs a single pulse (high -> low) on the configured pin.
  * This function will always wait for a previous call to finish.
