Index: LilyGo-EPD47/src/libjpeg/libjpeg.c
===================================================================
--- LilyGo-EPD47.orig/src/libjpeg/libjpeg.c
+++ LilyGo-EPD47/src/libjpeg/libjpeg.c
@@ -19,6 +19,8 @@
 #include "rom/tjpgd.h"
 #endif
 
+#include "py/runtime.h"
+
 #include <stdio.h>
 #include <string.h>
 #include <math.h>
@@ -126,7 +128,7 @@ const char *TAG = "LIB_JPEG";
 void libjpeg_init(void)
 {
     //解码图像内存申请， 来自PSRAM
-    decoded_image = (uint8_t *)heap_caps_malloc(EPD_WIDTH * EPD_HEIGHT, MALLOC_CAP_SPIRAM);
+    decoded_image = (uint8_t *)m_malloc(EPD_WIDTH * EPD_HEIGHT);
     if (decoded_image == NULL)
     {
         ESP_LOGE(TAG, "Initial alloc decoded_image failed!");
@@ -222,7 +224,7 @@ void show_jpg_from_buff(uint8_t *buff, u
 
 void libjpeg_deinit(void)
 {
-    free(decoded_image);
+    m_free(decoded_image);
 }
 
 /******************************************************************************/
