Index: micropython/ports/rp2/Makefile
===================================================================
--- micropython.orig/ports/rp2/Makefile
+++ micropython/ports/rp2/Makefile
@@ -30,6 +30,10 @@ ifdef MICROPY_BOARD_DIR
 	CMAKE_ARGS += -D MICROPY_BOARD_DIR=$(MICROPY_BOARD_DIR)
 endif
 
+ifdef PICO_SDK_PATH_OVERRIDE
+	CMAKE_ARGS += -D PICO_SDK_PATH_OVERRIDE=$(PICO_SDK_PATH_OVERRIDE)
+endif
+
 all:
 	[ -e $(BUILD)/CMakeCache.txt ] || cmake -S . -B $(BUILD) -DPICO_BUILD_DOCS=0 ${CMAKE_ARGS}
 	$(MAKE) $(MAKESILENT) -C $(BUILD)
@@ -37,7 +41,7 @@ all:
 clean:
 	$(RM) -rf $(BUILD)
 
-GIT_SUBMODULES += lib/mbedtls lib/pico-sdk lib/tinyusb
+GIT_SUBMODULES += lib/mbedtls lib/tinyusb
 
 submodules:
 ifeq ($(BOARD),PICO)
